zerver/actions/user_groups.py

```diff
@@ -175,7 +176,7 @@ def do_send_create_user_group_event(
             id=user_group.id,
             is_system_group=user_group.is_system_group,
             direct_subgroup_ids=[direct_subgroup.id for direct_subgroup in direct_subgroups],
-            can_mention_group=user_group.can_mention_group_id,
+            can_mention_group=get_group_setting_value_for_api(user_group.can_mention_group),
```

Comment:
This hunk might belong in the previous commit?

Reply:
Before this commit, one could only set the setting to named user groups while creating groups, so passing ID would have been fine then. But this commit adds support to set the setting to anonymous groups as well, and thus added this change here.